src/filter-image/FilterImage.tsx

```diff
@@ -19,13 +26,32 @@ const getFilters = (filters: FilterImageProps['filters']) => {
   });
 };
 
+const getFilterName = (filter: Filters) => {
+  return (
+    'filter' +
+    filter.reduce(
+      (acc, curr) =>
+        acc +
+        curr.name +
+        (Array.isArray(curr.values) ? curr.values.join(',') : curr.values) +
+        ';',
+      ''
+    )
+  );
+};
+
 export const FilterImage = (props: FilterImageProps) => {
   const { source, style, ...imageProps } = props;
+  const hasFilter = props.filters.length > 0;
+  const filterId = useMemo(() => getFilterName(props.filters), [props.filters]);
```

Comment:
What if all the filters are the same on both images? Wouldn't it be better to just use some hash and add it?

Reply:
Sure, any suggestions on how to get a relatively unique hash? I'm afraid that `Math.random` won't be sufficient.

Reply:
Done in [26cbf53](https://github.com/software-mansion/react-native-svg/pull/2346/commits/26cbf53ee2f9566c98eb2d8afc729eb4d9a16290)